assets/src/settings-page/supported-templates.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * WordPress dependencies
+ */
+import { useContext, useEffect, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { SupportedTemplatesToggle } from '../components/supported-templates-toggle';
+import { Options } from '../components/options-context-provider';
+import { Selectable } from '../components/selectable';
+import { SupportedTemplatesVisibility } from './supported-templates-visibility';
+
+/**
+ * Supported templates section of the settings page.
+ */
+export function SupportedTemplates() {
+	const { editedOptions } = useContext( Options );
+
+	const { reader_theme: readerTheme, theme_support: themeSupport } = editedOptions;
+
+	const supportedTemplatesContainer = useRef();
+
+	/**
+	 * Pull the PHP-generated form inputs into this React-generated section.
+	 */
+	useEffect( () => {
+		const settingsSections = [ ...document.querySelectorAll( '#amp-settings-sections > table' ) ];
+		const supportedTemplatesTable = settingsSections.find( ( section ) => section.querySelector( '.amp-template-support-field' ) );
+		if ( supportedTemplatesTable ) {
+			supportedTemplatesContainer.current.appendChild( supportedTemplatesTable );
+		}
+	}, [] );
+
+	return (
+		<section className={ ( 'reader' === themeSupport && 'legacy' === readerTheme ) || ! themeSupport ? 'hidden' : '' }>
```

Comment:
This section should only be hidden in legacy Reader mode.

```suggestion
		<section className={ 'reader' === themeSupport && 'legacy' === readerTheme ? 'hidden' : '' }>
```